docs/guides/integration/aws-lambda.md

```diff
@@ -0,0 +1,407 @@
+# Using uv with AWS Lambda
+
+[AWS Lambda](https://aws.amazon.com/lambda/) is a serverless computing service that lets you run
+code without provisioning or managing servers.
+
+You can use uv with AWS Lambda to manage your Python dependencies, build your deployment package,
+and deploy your Lambda functions.
+
+!!! tip
+
+    Check out the [`uv-aws-lambda-example`](https://github.com/astral-sh/uv-aws-lambda-example) project for
+    an example of best practices when using uv to deploy an application to AWS Lambda.
+
+## Getting started
+
+To start, assume we have a minimal FastAPI application with the following structure:
+
+```plaintext
+project
+├── pyproject.toml
+└── app
+    ├── __init__.py
+    └── main.py
+```
+
+Where the `pyproject.toml` contains:
+
+```toml title="pyproject.toml"
+[project]
+name = "uv-aws-lambda-example"
+version = "0.1.0"
+requires-python = ">=3.13"
+dependencies = [
+    # FastAPI is a modern web framework for building APIs with Python.
+    "fastapi",
+    # Mangum is a library that adapts ASGI applications to AWS Lambda and API Gateway.
+    "mangum",
+]
+
+[dependency-groups]
+dev = [
+    # In development mode, include the FastAPI development server.
+    "fastapi[standard]",
+]
+```
+
+And the `main.py` file contains:
+
+```python title="app/main.py"
+import logging
+
+from fastapi import FastAPI
+from mangum import Mangum
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+app = FastAPI()
+handler = Mangum(app)
+
+
+@app.get("/")
+async def root() -> str:
+    return "Hello, world!"
+```
+
+We can run this application locally with:
+
+```console
+$ uv run fastapi dev
+```
+
+From there, opening http://127.0.0.1:8000/ in a web browser will display "Hello, world!"
+
+## Deploying a Docker image
+
+To deploy to AWS Lambda, we need to build a container image that includes the application code and
+dependencies in a single output directory.
+
+We'll follow the principles outlined in the [Docker guide](./docker.md) (in particular, a
+multi-stage build) to ensure that the final image is as small and cache-friendly as possible.
+
+In the first stage, we'll populate a single directory with all application code and dependencies. In
+the second stage, we'll copy this directory over to the final image, omitting the build tools and
+other unnecessary files.
+
+```dockerfile title="Dockerfile"
+FROM ghcr.io/astral-sh/uv:latest AS uv
```

Comment:
Here, we _could_ pin to the current uv version and add this to the files that are updated on release.